react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const items = await readItems();
+        setItems(items);
+      } catch (err) {
+        setIsLoading(false);
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchData();
+  }, [setIsLoading]);
```

Comment:
You are correct that you are directly using setIsLoading, so you would think it should be listed as a dependency. However, React guarantees that state setters never change, so it isn't necessary to list it.